lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/TrinoJsonFactory.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import com.fasterxml.jackson.core.TSFBuilder;
+import com.fasterxml.jackson.core.io.CharacterEscapes;
+import com.fasterxml.jackson.core.io.InputDecorator;
+import com.fasterxml.jackson.core.io.OutputDecorator;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+public final class TrinoJsonFactory
```

Comment:
https://github.com/trinodb/trino/pull/17950 is merged. please rebase and remove this class

Reply:
Rebased but I'm not sure whether allowing JsonFactory mutation is a right thing to do. It's error-prone as it can happen in the runtime, and there is no mechanism that will prohibit cross-class reuse of the existing factories. Ideally I'd delegate that to the airlift and guice to both allow configuring and injecting factories but it's a bigger change as some of those classes are not even DI managed.